src/linkify-matrix.ts

```diff
@@ -132,77 +147,44 @@ function matrixLinkify(linkify): void {
     S_USERID.on(TT.DOT, S_AT_NAME_COLON_DOMAIN_DOT); // accept repeated TLDs (e.g .org.uk)
     S_USERID.on(TT.COLON, S_USERID_COLON); // do not accept trailing `:`
     S_USERID_COLON.on(TT.NUM, S_USERID_COLON_NUM); // but do accept :NUM (port specifier)
-
-    const GROUPID = function(value) {
-        MultiToken.call(this, value);
-        this.type = 'groupid';
-        this.isLink = true;
-    };
-    GROUPID.prototype = new MultiToken();
-
-    const S_PLUS = S_START.jump(TT.PLUS);
-    const S_PLUS_NAME = new linkify.parser.State();
-    const S_PLUS_NAME_COLON = new linkify.parser.State();
-    const S_PLUS_NAME_COLON_DOMAIN = new linkify.parser.State(GROUPID);
-    const S_PLUS_NAME_COLON_DOMAIN_DOT = new linkify.parser.State();
-    const S_GROUPID = new linkify.parser.State(GROUPID);
-    const S_GROUPID_COLON = new linkify.parser.State();
-    const S_GROUPID_COLON_NUM = new linkify.parser.State(GROUPID);
-
-    const groupIdTokens = [
-        TT.DOT,
-        TT.UNDERSCORE,
-        TT.PLUS,
-        TT.NUM,
-        TT.DOMAIN,
-        TT.TLD,
-
-        // as in roomnameTokens
-        TT.LOCALHOST,
-    ];
-
-    S_PLUS.on(groupIdTokens, S_PLUS_NAME);
-    S_PLUS_NAME.on(groupIdTokens, S_PLUS_NAME);
-    S_PLUS_NAME.on(TT.DOMAIN, S_PLUS_NAME);
-
-    S_PLUS_NAME.on(TT.COLON, S_PLUS_NAME_COLON);
-
-    S_PLUS_NAME_COLON.on(TT.DOMAIN, S_PLUS_NAME_COLON_DOMAIN);
-    S_PLUS_NAME_COLON.on(TT.LOCALHOST, S_GROUPID); // accept +foo:localhost
-    S_PLUS_NAME_COLON.on(TT.TLD, S_GROUPID); // accept +foo:com (mostly for (TLD|DOMAIN)+ mixing)
-    S_PLUS_NAME_COLON_DOMAIN.on(TT.DOT, S_PLUS_NAME_COLON_DOMAIN_DOT);
-    S_PLUS_NAME_COLON_DOMAIN_DOT.on(TT.DOMAIN, S_PLUS_NAME_COLON_DOMAIN);
-    S_PLUS_NAME_COLON_DOMAIN_DOT.on(TT.TLD, S_GROUPID);
-
-    S_GROUPID.on(TT.DOT, S_PLUS_NAME_COLON_DOMAIN_DOT); // accept repeated TLDs (e.g .org.uk)
-    S_GROUPID.on(TT.COLON, S_GROUPID_COLON); // do not accept trailing `:`
-    S_GROUPID_COLON.on(TT.NUM, S_GROUPID_COLON_NUM); // but do accept :NUM (port specifier)
 }
 
-// stubs, overwritten in MatrixChat's componentDidMount
-matrixLinkify.onUserClick = function(e: MouseEvent, userId: string) { e.preventDefault(); };
-matrixLinkify.onAliasClick = function(e: MouseEvent, roomAlias: string) { e.preventDefault(); };
-matrixLinkify.onGroupClick = function(e: MouseEvent, groupId: string) { e.preventDefault(); };
+function onUserClick(e: MouseEvent, userId: string) {
+    const member = new RoomMember(null, userId);
+    if (!member) { return; }
+    dis.dispatch<ViewUserPayload>({
+        action: Action.ViewUser,
+        member: member,
+    });
+}
+function onAliasClick(e: MouseEvent, roomAlias: string) {
+    event.preventDefault();
+    dis.dispatch({ action: 'view_room', room_alias: roomAlias });
+}
+function onGroupClick(e: MouseEvent, groupId: string) {
+    event.preventDefault();
+    dis.dispatch({ action: 'view_group', group_id: groupId });
+}
```

Comment:
`event` got renamed to `e` but the preventDefault is still called on the old identifier

Reply:
Fixed